packages/lib/services/searchengine/SearchEngineUtils.ts

```diff
@@ -49,9 +50,24 @@ export default class SearchEngineUtils {
 		// issue: https://discourse.joplinapp.org/t/how-to-recover-corrupted-database/9367
 		if (noteIds.length !== notes.length) {
 			// remove null objects
-			return sortedNotes.filter(n => n);
+      const results2:any = [];
+      const filtered_notes = sortedNotes.filter(n => n);
+      for(let i=0;i<filtered_notes.length;i++){
+        if(!Setting.value('showCompletedTodos') && filtered_notes[i].todo_completed){
```

Comment:
You don't need to check setting on every iteration. Maybe the runtime is smart enough to hoist if out of the loop but then, maybe it's not.

Reply:
In fact, if showCompletedTodos is enabled you don't need the loop at all.

Reply:
The searchengie traverses through all the notes regardless of them being completedTodos even if the showCompletedTodos option is toggled, hence we need a for loop to filter through the search results right before they are passed on to the notelist renderer

Reply:
You only need the loop if showCompletedTodos is on, otherwise the whole loop is a no-op. And even if you do need the loop, you only need to read the setting once at the start as it can't change in the middle of the loop.

Reply:
oh i see, so i need to run the loop only if the setting is toggled, reformat the code and change the array type to any[ ], but do you think the method I used here to fix the bug is valid?
Also I couldn't understand why the check failed, was it because I didn't format the code correctly?

Reply:
Have you considered doing the check in sql instead?

Reply:
>  Also I couldn't understand why the check failed, was it because I didn't format the code correctly?

Yes, i'm pretty sure that's the reason.

Reply:
Im talking about the Travis CI check, not sure what doing a check in SQL refers to.